packages/backend/src/invoice/routes.ts

```diff
@@ -0,0 +1,147 @@
+  config: IAppConfig
+  logger: Logger
+  accountService: AccountService
+  invoiceService: InvoiceService
+  streamServer: StreamServer
+}
+
+export interface InvoiceRoutes {
+  get(ctx: AppContext): Promise<void>
+  create(ctx: AppContext): Promise<void>
+}
+
+export function createInvoiceRoutes(deps_: ServiceDependencies): InvoiceRoutes {
+  const logger = deps_.logger.child({
+    service: 'InvoiceRoutes'
+  })
+  const deps = { ...deps_, logger }
+  return {
+    get: (ctx: AppContext) => getInvoice(deps, ctx),
+    create: (ctx: AppContext) => createInvoice(deps, ctx)
+  }
+}
+
+// Spec: https://docs.openpayments.dev/invoices#get
+async function getInvoice(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  const { invoiceId } = ctx.params
+  ctx.assert(validateId(invoiceId), 400, 'invalid id')
+  const acceptJSON = ctx.accepts('application/json')
+  const acceptStream = ctx.accepts('application/ilp-stream+json')
+  ctx.assert(acceptJSON || acceptStream, 406)
+
+  const invoice = await deps.invoiceService.get(invoiceId)
+  if (!invoice) return ctx.throw(404)
+
+  const amountReceived = await deps.accountService.getBalance(invoice.accountId)
+  if (amountReceived === undefined) {
+    deps.logger.error(
+      { account: invoice.accountId, invoice: invoice.id },
+      'balance not found'
+    )
+    return ctx.throw(500)
+  }
+
+  const body = invoiceToBody(deps, invoice, amountReceived)
+  ctx.body = body
+  if (!acceptStream) return
+
+  const { ilpAddress, sharedSecret } = deps.streamServer.generateCredentials({
+    paymentTag: invoice.accountId,
+    // TODO receipt support on invoices?
+    //receiptSetup:
+    //  nonce && secret
+    //    ? {
+    //        nonce: Buffer.from(nonce.toString(), 'base64'),
+    //        secret: Buffer.from(secret.toString(), 'base64')
+    //      }
+    //    : undefined,
+    asset: {
+      code: invoice.account.asset.code,
+      scale: invoice.account.asset.scale
+    }
+  })
+
+  body['ilpAddress'] = ilpAddress
+  body['sharedSecret'] = sharedSecret
+}
+
+// Spec: https://docs.openpayments.dev/invoices#create
+async function createInvoice(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  const { paymentPointerId } = ctx.params
+  ctx.assert(validateId(paymentPointerId), 400, 'invalid payment pointer')
+  ctx.assert(ctx.accepts('application/json'), 406, 'must accept json')
+  ctx.assert(
+    ctx.get('Content-Type') === 'application/json',
+    400,
+    'must send json body'
+  )
+
+  const { body } = ctx.request
+  if (typeof body !== 'object') return ctx.throw(400, 'json body required')
+  const amountToReceive = tryParseAmount(body['amount'])
+  if (!amountToReceive) return ctx.throw(400, 'invalid amount')
+  const expiresAt = Date.parse(body['expiresAt'] as string)
+  if (!expiresAt) return ctx.throw(400, 'invalid expiresAt')
+  if (typeof body.description !== 'string')
+    return ctx.throw(400, 'invalid description')
```

Comment:
Do we want to make `description` optional?
https://docs.openpayments.dev/invoices#create